### PR TITLE
OnnxToTorch ReduceMax lowering

### PR DESCRIPTION
**Pull request by aldesilv**
https://github.com/nod-ai/SHARK-Turbine/issues/352

### CODE REVIEW

File: test/Conversion/TorchOnnxToTorch/simple_ops_q_to_z.mlir
Comment (wu-s-john) [inline test/Conversion/TorchOnnxToTorch/simple_ops_q_to_z.mlir:536]: What's unique about `Onnx::ReduceMax` is that it can run a reduce operation on mutliple axes. Whereas for `Torch::max` runs the reduce operation on just one axes.

Can you construct a test where you test this on multiple axes?

Comment (wu-s-john) [reply test/Conversion/TorchOnnxToTorch/simple_ops_q_to_z.mlir:536]: NIT, I think `arg1` should be constructed from a literal tensor (or a tensor created from a function like `arange`). It would be easier to understand the test.

File: test/Conversion/TorchOnnxToTorch/simple_ops_q_to_z.mlir
Comment (wu-s-john) [inline test/Conversion/TorchOnnxToTorch/simple_ops_q_to_z.mlir:541]: NIT: for tests, please give an alias to each variable. It would be easier if you can assign a variable key instructions.

Like for `torch.aten.select.int %arg1, %int0, %int0_0`, it can be assigned as 

```
[[SELECTED_DIM:*]]= torch.aten.select.int %arg1, %int0, %int0_0
```

**Review by wu-s-john — CHANGES_REQUESTED** [review]
Looks good overall.

Just need to make some small improvements on test. Please tag me on discord to review your code again. My handle is @phantom_type on the llvm discord channel.

**Review by wu-s-john — APPROVED** [review]
Looks good to me.

You should probably get another approval from a reviewer since I'm not a maintainer of the codebase.

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp
Comment (vivekkhandelwal1) [inline lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp:530]: You can define both of these on the same line.

**Review by vivekkhandelwal1 — APPROVED** [review]
Just a small nit, rest LGTM.

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp
Comment (vivekkhandelwal1) [reply lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp:530]: Also, please `clang-format` if not done so.

**Pull request merged**
